src/wallet/wallet.cpp

```diff
@@ -2819,8 +2819,12 @@ std::shared_ptr<CWallet> CWallet::Create(WalletContext& context, const std::stri
             warnings.push_back(strprintf(_("Error reading %s! Transaction data may be missing or incorrect."
                                            " Rescanning wallet."), walletFile));
             rescan_required = true;
-        }
-        else {
+        } else if (nLoadWalletRet == DBErrors::UNKNOWN_DESCRIPTOR) {
+            error = strprintf(_("Error loading %s: Unrecognized descriptor found.\n\n"
+                                "The wallet might had been created on a newer version. Downgrading is incompatible with descriptor wallets\n"
```

Comment:
I don't think it is correct to say `Downgrading is incompatible with descriptor wallets`. Not all downgrading is incompatible, just this particular one with new descriptors. I think you should just leave that sentence out.

Reply:
I wrote it in that way because at this point of the flow we already tried to parse the descriptor and failed, knowing for sure that either the downgrade is incompatible or the db is corrupted.
But sure, will remove the line.